src/utils/daos_autotest.c

```diff
@@ -778,7 +802,79 @@ kv_insertaux(void)
 	if (rc)
 		D_GOTO(fail_close, rc);
 
-	oh = DAOS_HDL_INVAL;
+	step_success("");
+	return 0;
+
+skip_step:
+	step_skip("Group size(2) is larger than domain_nr(%d)", domain_nr);
+	return 0;
+
+fail_open:
+	step_fail("failed to open object: %s", d_errdesc(rc));
+	return -1;
+
+fail_insert:
+	step_fail("failed to insert: %s", d_errdesc(put_rc));
+	return -1;
+
+fail_close:
+	step_fail("failed to close object: %s", d_errdesc(rc));
+	return -1;
```

Comment:
Maybe better to keep only one exit label and move error messages near to goto with if condition to easily verify the message is corresponds the error which happens. I think it's difficult to maintain many labels and always go down to see the error message which prints.

Reply:
Great comments and feedback. I'm always interested in increasing maintainability and readability in code. This should be all set